zkevm-circuits/src/gadget.rs

```diff
@@ -15,6 +15,21 @@ impl<T, F: FieldExt> Variable<T, F> {
     }
 }
 
+pub(crate) mod comparator;
 pub(crate) mod evm_word;
+pub(crate) mod is_equal;
 pub(crate) mod is_zero;
+pub(crate) mod less_than;
 pub(crate) mod monotone;
+
+/// Restrict an expression to be a boolean.
+pub fn bool_check<F: FieldExt>(value: Expression<F>) -> Expression<F> {
+    range_check(value, 2)
+}
+
+/// Restrict an expression such that 0 <= word < range.
+pub fn range_check<F: FieldExt>(word: Expression<F>, range: usize) -> Expression<F> {
+    (1..range).fold(word.clone(), |acc, i| {
+        acc * (Expression::Constant(F::from(i as u64)) - word.clone())
+    })
+}
```

Comment:
Be careful with this as depending on the range, a lookup might be a much better option.

Reply:
> I think both gadgets developed here should go into the `gadgets` workspace member.
> 
> As they could be re-used by more than one circuit and also are not EVMWord-focused.

I agree, but the `LtChip` uses one of the utils, i.e. `from_bytes`. There might be some code duplication if I declare such a function in `gadgets` crate again. What do you think about that?

Reply:
Hmmm that's a good point!!!

Maybe we can externalize quite a lot of the gadgets in `util`.
Will file a PR for it. So that we don't mix stuff in the same PR.